### PR TITLE
Adding optional title and description to images

### PR DESCRIPTION
**Pull request by pfwd**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Fixed tickets | #3720 |
| License | MIT |
| Doc PR | #3720 |

Work for adding title and description fields to images

### CODE REVIEW

File: src/Sylius/Component/Core/Model/Image.php [hunk 1]
```diff
@@ -118,6 +133,40 @@ public function setPath($path)
...
+        return $this->title;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setTitle($title)
+    {
+        $this->title = $title;
+        return $this;
```
Comment (pjedrzejewski) [inline src/Sylius/Component/Core/Model/Image.php]: We removed fluent interface.

File: src/Sylius/Component/Core/Model/Image.php [hunk 2]
```diff
@@ -35,6 +36,20 @@ class Image implements ImageInterface
     protected $path;
 
     /**
+     * Image title
```
Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Model/Image.php]: Redundant comment.

Comment (pfwd) [inline src/Sylius/Component/Core/Model/Image.php]: Thats strange.  I don't see this in the code:
https://github.com/pfwd/Sylius/blob/46f41758f5f2ccd02308fa1887c3b3179fa7137a/src/Sylius/Component/Core/Model/Image.php

Comment (michalmarcinkowski) [inline src/Sylius/Component/Core/Model/Image.php]: I meant unnecessary, cause it's obvious. Like all of them in this class, e.g.
```php
/**
 * Id
 *
```

Comment (pfwd) [inline src/Sylius/Component/Core/Model/Image.php]: @michalmarcinkowski Agreed, I've removed the comments

File: src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 class ImageType extends AbstractResourceType
 {
+
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php]: This should be removed.

File: src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php [hunk 4]
```diff
@@ -24,6 +25,14 @@ public function buildForm(FormBuilderInterface $builder, array $options)
         $builder->add('file', 'file', array(
             'label' => 'sylius.form.image.file'
         ));
+        $builder->add('title', 'text', array(
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php]: You could use fluent interface as in other forms.

File: src/Sylius/Bundle/CoreBundle/Resources/config/doctrine/model/ProductVariantImage.orm.xml [hunk 5]
```diff
@@ -18,6 +18,8 @@
 
     <mapped-superclass name="Sylius\Component\Core\Model\ProductVariantImage" table="sylius_product_variant_image">
         <many-to-one field="variant" target-entity="Sylius\Component\Product\Model\VariantInterface" inversed-by="images" />
+        <field name="title" column="title" type="string" nullable="true" />
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Resources/config/doctrine/model/ProductVariantImage.orm.xml]: IMO fields should be declared before relations, same as it's done in other model declarations.

File: src/Sylius/Component/Core/Model/Image.php [hunk 6]
```diff
@@ -13,37 +13,38 @@
 
 class Image implements ImageInterface
 {
+
```
Comment (stloyd) [inline src/Sylius/Component/Core/Model/Image.php]: Should be reverted.

**Comment by stloyd** [comment]
Migration file is missing ;)

File: src/Sylius/Component/Core/Model/Image.php [hunk 7]
```diff
@@ -118,6 +117,38 @@ public function setPath($path)
     /**
      * {@inheritdoc}
      */
+    public function getTitle()
```
Comment (NeverResponse) [inline src/Sylius/Component/Core/Model/Image.php]: Add the `title` and `description` methods to the `ImageInterface`

Comment (pjedrzejewski) [inline src/Sylius/Component/Core/Model/Image.php]: :+1:

Comment (pfwd) [inline src/Sylius/Component/Core/Model/Image.php]: Other classes rely on the ImageInterface and not all need a title and description
The Taxon model is a example:

    class Taxon extends BaseTaxon implements ImageInterface, TaxonInterface

File: src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php (hunk 3 above)
Comment (pfwd) [inline src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php]: Is there a specific format I should use in my ide (Netbeans)?

**Comment by pfwd** [comment]
@stloyd @pjedrzejewski I have tweak it with your recommendations.

File: src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php [hunk 8]
```diff
@@ -16,13 +16,22 @@
...
 {
+
     /**
      * {@inheritdoc}
      */
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder->add('file', 'file', array(
-            'label' => 'sylius.form.image.file'
+                    'label' => 'sylius.form.image.file'
```
Comment (stloyd) [inline src/Sylius/Bundle/CoreBundle/Form/Type/ImageType.php]: Indent is wrong, should be ~8 spaces less

**Comment by pfwd** [comment]
@pjedrzejewski Not sure what has gone wrong here.  I was having issues rebasing the branch as I had some conflicts in files that didn't intentionally alter.
Perhaps it would be best If I closed this branch down and started again. It's only a few file changes. 
What do you think is best course of action

**Pull request closed without merging**